runtime/compiler/env/j9methodServer.cpp

```diff
@@ -324,21 +324,27 @@ TR_ResolvedJ9JITServerMethod::getResolvedPossiblyPrivateVirtualMethod(TR::Compil
 #else
    auto compInfoPT = (TR::CompilationInfoPerThreadRemote *) _fe->_compInfoPT;
    TR_ResolvedMethod *resolvedMethod = NULL;
-   if (compInfoPT->getCachedResolvedMethod(compInfoPT->getResolvedMethodKey(TR_ResolvedMethodType::VirtualFromCP, (TR_OpaqueClassBlock *) _ramClass, cpIndex), this,
-                                           &resolvedMethod, unresolvedInCP))
+   TR_ResolvedMethodKey key = compInfoPT->getResolvedMethodKey(TR_ResolvedMethodType::VirtualFromCP, (TR_OpaqueClassBlock *) _ramClass, cpIndex);
+   if (!((_fe->_compInfoPT->getClientData()->getRtResolve()) &&
```

Comment:
Yes, I think this is a bug fix, because in some cases, the resolved possibly private virtual method can be cached from outside a call to `getResolvedPossiblyPrivateVirtualMethod`, e.g. in `cacheResolvedMethodsCallees`. There, we prefetch resolved methods before they are actually requested, so we do not know the value of `ignoreRtResolve`  that will be used when the method will be retrieved from the cache, so it's assumed to be `true`.
The value of `shouldCompileTimeResolve` shouldn't matter because the client-side call to `getResolvedPossiblyPrivateVirtualMethod` already computes that value internally and thus resolves/does not resolve correctly. Thinking about it,  we actually only need to check `ignoreRtResolve` on the server, checks of `getRtResolve` shouldn't matter for the same reason `shouldCompileTimeResolve` value does not matter.